src/ui/base-page-login.js

```diff
@@ -0,0 +1,371 @@
+/*
+For copyright information, see the AUTHORS.md file in the docs directory of this distribution and at
+https://github.com/fluid-project/sjrk-story-telling/blob/main/docs/AUTHORS.md
+
+Licensed under the New BSD license. You may not use this file except in compliance with this licence.
+You may obtain a copy of the BSD License at
+https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/main/LICENSE.txt
+*/
+
+"use strict";
+
+(function ($, fluid) {
+
+    // The login page base grade
+    fluid.defaults("sjrk.storyTelling.base.page.login", {
+        gradeNames: ["sjrk.storyTelling.base.page.withAuthorControls"],
+        model: {
+            // loginState can be one of the following values:
+            // "ready" (the initial state), "requestSent", "responseReceived"
+            loginState: "ready",
+            loginButtonDisabled: false,
+            progressAreaVisible: false,
+            serverResponseText: false,
+            emailErrorMessage: "",
+            passwordErrorMessage: ""
+        },
+        modelListeners: {
+            progressAreaVisible: {
```

Comment:
Recently, I learnt a a simpler way to control the visibility of a dom element with the new renderer via "modelRelay" block:

```
modelRelay: {
    progressAreaVisibleChange: {
        source: "progressAreaVisible",
        target: "{loginUi}.dom.progressArea.visible"
    }
}
```

Just for your information. You don't need to make the change.

Reply:
@cindyli thanks for this info. We've updated to the FLUID-6580 line of Infusion so I guess this should be do-able now. @BlueSlug, if you decide to do this change later, could you please file a JIRA about it.

Reply:
I knew this was on its way, and it looks incredibly convenient! This would have applications in other parts of the codebase, too.

I'll see how quick it is to implement in this PR, and if it takes more than an hour or so I'll defer it to another Jira.

Reply:
I was unable to get it working, nor was I able to find any sample code (could have sworn I saw some in another project!) or documentation. I have filed [SJRK-451](https://issues.fluidproject.org/browse/SJRK-451) to tackle this update some other time.